internal/types/kruizePayload/common.go

```diff
@@ -91,103 +91,160 @@ func convertMetricToString(data interface{}) string {
+					Sum:    sum,
+					Avg:    avg,
+					Format: "cores",
 				},
 			},
-			{
-				Name: "cpuThrottle",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Max:    convertMetricToString(c["cpu_throttle_container_max_MAX"]),
-						Sum:    convertMetricToString(c["cpu_throttle_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["cpu_throttle_container_avg_MEAN"]),
-						Format: "cores",
-					},
+		})
+	}
+
+	// cpuThrottle
+	sum = convertMetricToString(c["cpu_throttle_container_sum_SUM"])
+	avg = convertMetricToString(c["cpu_throttle_container_avg_MEAN"])
+	max = convertMetricToString(c["cpu_throttle_container_max_MAX"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "cpuThrottle",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Max:    max,
+					Sum:    sum,
+					Avg:    avg,
+					Format: "cores",
 				},
 			},
-			{
-				Name: "memoryRequest",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Sum:    convertMetricToString(c["memory_request_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["memory_request_container_avg_MEAN"]),
-						Format: "MiB",
-					},
+		})
+	}
+
+	// memoryRequest
+	sum = convertMetricToString(c["memory_request_container_sum_SUM"])
+	avg = convertMetricToString(c["memory_request_container_avg_MEAN"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "memoryRequest",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Sum:    sum,
+					Avg:    avg,
+					Format: "MiB",
 				},
 			},
-			{
-				Name: "memoryLimit",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Sum:    convertMetricToString(c["memory_limit_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["memory_limit_container_avg_MEAN"]),
-						Format: "MiB",
-					},
+		})
+	}
+
+	// memoryLimit
+	sum = convertMetricToString(c["memory_limit_container_sum_SUM"])
+	avg = convertMetricToString(c["memory_limit_container_avg_MEAN"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "memoryLimit",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Sum:    sum,
+					Avg:    avg,
+					Format: "MiB",
 				},
 			},
-			{
-				Name: "memoryUsage",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Min:    convertMetricToString(c["memory_usage_container_min_MIN"]),
-						Max:    convertMetricToString(c["memory_usage_container_max_MAX"]),
-						Sum:    convertMetricToString(c["memory_usage_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["memory_usage_container_avg_MEAN"]),
-						Format: "MiB",
-					},
+		})
+	}
+
+	// memoryUsage
+	sum = convertMetricToString(c["memory_usage_container_sum_SUM"])
+	avg = convertMetricToString(c["memory_usage_container_avg_MEAN"])
+	min = convertMetricToString(c["memory_usage_container_min_MIN"])
+	max = convertMetricToString(c["memory_usage_container_max_MAX"])
```

Comment:
@patilsuraj767 Shall we check for `min` and `max` on if conditions for each metric append?

Reply:
I just thought that it might happen for some reason we might have `sum` and `avg` but do not have `min` and `max`.  As far as I know sum and avg is a mandatory so I added condition only on `sum` and `avg`.